pallets/asset/src/lib.rs

```diff
@@ -1788,15 +1788,23 @@ impl<T: Config> Module<T> {
             asset_type: asset_type.clone(),
         };
         Tokens::insert(&ticker, token);
-        AssetNames::insert(&ticker, name);
+        AssetNames::insert(&ticker, name.clone());
```

Comment:
```suggestion
        AssetNames::insert(&ticker, &name);
```
It should be possible to avoid cloning here.  The `insert` method can take a reference or value here, it just needs to encode the value and not take ownership.

For a normal Rust `HashMap` or other container type, this wouldn't be possible.  But Substrate storage types don't really hold the value, then encode it to raw bytes before writing it to the storage backend.

Reply:
just made the change.